doc/api/.eslintrc.yaml

```diff
@@ -0,0 +1,5 @@
+overrides:
+  - files:
+    - "./esm.md"
+    rules:
+      dot-notation: 0
```

Comment:
Can you please explain this? Is this file needed?

Reply:
Yes, sorry for the lack of explanation. If you look at https://github.com/nodejs/node/pull/33563/commits/7e01cc806a8fa6a5f51d53aa87cd1087595e5f44, you will see that it is changing &hellip;

```js
exports.customExportName.set('value');
``` 
&hellip; to &hellip;
```js
exports['customExportName'].set('value');
```

&hellip; essentially doing the same thing. The reason for this is as a workaround to [a warning in Closure Compiler](https://closure-compiler.appspot.com/home#code%3D%252F%252F%2520%253D%253DClosureCompiler%253D%253D%250A%252F%252F%2520%2540compilation_level%2520ADVANCED_OPTIMIZATIONS%250A%252F%252F%2520%2540warning_level%2520VERBOSE%250A%252F%252F%2520%2540jscomp_error%2520strictCheckTypes%250A%252F%252F%2520%2540language_out%2520ECMASCRIPT_NEXT%250A%252F%252F%2520%2540checks_only%250A%252F%252F%2520%253D%253D%252FClosureCompiler%253D%253D%250A%250A%252F**%250A%2520*%2520%2540param%2520%257Bstring%257D%2520url%250A%2520*%2520%2540returns%2520%257BPromise%253C%257B%250A%2520*%2520%2520%2520exports%253A%2520!(Array%253Cstring%253E)%252C%250A%2520*%2520%2520%2520execute%253A%2520Function%252C%250A%2520*%2520%257D%253E%257D%2520response%250A%2520*%252F%250Aexport%2520async%2520function%2520dynamicInstantiate(url)%2520%257B%250A%2520%2520return%2520%257B%250A%2520%2520%2520%2520exports%253A%2520%255B'customExportName'%255D%252C%250A%2520%2520%2520%2520execute%253A%2520%252F**%2520%2540param%2520%257B%253F%257D%2520exports%2520*%252F%2520(exports)%2520%253D%253E%2520%257B%250A%2520%2520%2520%2520%2520%2520%252F%252F%2520Get%2520and%2520set%2520functions%2520provided%2520for%2520pre-allocated%2520export%2520names%250A%2520%2520%2520%2520%2520%2520exports.customExportName.set('value')%253B%250A%2520%2520%2520%2520%257D%252C%250A%2520%2520%257D%253B%250A%257D) — possible inexistant property. Since the goal is to have this code not warn or error by either compiler, I've had to make this slight change, which actually makes it even more clear IMO. The problem w/ doing this is that [the `dot-notation` ESLint rule](https://eslint.org/docs/rules/dot-notation) is enabled and causes the CI to fail. We want to ignore this rule, but we can't use disable directives (`/* eslint-disable */`), because that would show up in our snippet, so we disable this rule for the file via this override.

Reply:
Seems like it's related to https://github.com/nodejs/node/pull/33563/files#diff-8e67f407bc32a0569e25d7ecaff6e494R1366

I personally would keep that as it was before and remove this override again.

Reply:
Not sure I understand, [look at what happens in Closure](https://closure-compiler.appspot.com/home#code%3D%252F%252F%2520%253D%253DClosureCompiler%253D%253D%250A%252F%252F%2520%2540compilation_level%2520ADVANCED_OPTIMIZATIONS%250A%252F%252F%2520%2540warning_level%2520VERBOSE%250A%252F%252F%2520%2540jscomp_error%2520strictCheckTypes%250A%252F%252F%2520%2540language_out%2520ECMASCRIPT_NEXT%250A%252F%252F%2520%2540checks_only%250A%252F%252F%2520%253D%253D%252FClosureCompiler%253D%253D%250A%250A%252F**%250A%2520*%2520%2540param%2520%257Bstring%257D%2520url%250A%2520*%2520%2540returns%2520%257BPromise%253C%257B%250A%2520*%2520%2520%2520exports%253A%2520!(Array%253Cstring%253E)%252C%250A%2520*%2520%2520%2520execute%253A%2520Function%252C%250A%2520*%2520%257D%253E%257D%2520response%250A%2520*%252F%250Aexport%2520async%2520function%2520dynamicInstantiate(url)%2520%257B%250A%2520%2520return%2520%257B%250A%2520%2520%2520%2520exports%253A%2520%255B'customExportName'%255D%252C%250A%2520%2520%2520%2520execute%253A%2520%252F**%2520%2540param%2520%257B%253F%257D%2520exports%2520*%252F%2520(exports)%2520%253D%253E%2520%257B%250A%2520%2520%2520%2520%2520%2520%252F%252F%2520Get%2520and%2520set%2520functions%2520provided%2520for%2520pre-allocated%2520export%2520names%250A%2520%2520%2520%2520%2520%2520exports.customExportName.set('value')%253B%250A%2520%2520%2520%2520%257D%252C%250A%2520%2520%257D%253B%250A%257D) by keeping that as it was.

```
JSC_POSSIBLE_INEXISTENT_PROPERTY: Property customExportName never defined on exports at line 13 character 14
      exports.customExportName.set('value');
```

This is no bueno.

Reply:
I'm not sure that “it needs to compile without warnings in Closure Compiler” is a worthwhile goal, at least not right now. Adding JSDoc annotations is definitely useful, but additionally targeting Closure Compiler and doing it right would mean integrating that into our CI linting and so on. That should maybe be its own PR that’s part of moving these examples into their own files outside of the `.md` files.

Reply:
> I'm not sure that “it needs to compile without warnings in Closure Compiler” is a worthwhile goal, at least not right now.

The actual goal is to ensure that they compile silently. Compiled by what? The TypeScript and Closure compilers. Silently means no warnings or errors.

> Adding JSDoc annotations is definitely useful, but additionally targeting Closure Compiler and doing it right would mean integrating that into our CI linting and so on.

Agreed, however, the likelihood of that happening is undefined. I've opted to keep the proofs for these snippets as permalinks in the commit messages. This should serve as a suitable stand-in until the ideal type checking facilities are in place.

---

Looking forward to https://github.com/nodejs/node/pull/33501 landing, at which point I will rebase to remove https://github.com/nodejs/node/commit/7e01cc806a8fa6a5f51d53aa87cd1087595e5f44 and  https://github.com/nodejs/node/pull/33563/commits/ddc1ab0db3fe3557651decd1f1c345c2f2294d92 resolving this conversation.

Reply:
I marked this as blocked until #33501 lands.

Reply:
It's unblocked again.